extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/RestAuthenticationHandler.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.vertx.http.runtime.security;
+
+import javax.enterprise.inject.spi.CDI;
+
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+public class RestAuthenticationHandler implements Handler<RoutingContext> {
```

Comment:
Hi Stuart @stuartwdouglas I wonder, if it the authentication method should be called as `RequestBodyAuthenticationMethod` (or similar ) so that we can also get the same properties from the form url encoded payload or some other format, depending on the media type ? If media type is missing, then simple JSON as in the PR is default payload format.
That said, I'm not sure if generalizing the extraction of the credentials from the payload makes much sense :-)